gradle-remote

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Please note that this script is quite basic and not recommended for use with public build servers.
+# Since rsync transfers all files from the developer's machine, including potentially sensitive information
+# like secrets or configuration files, it should be used with care.
+
+if ! [ -f build.gradle.kts ]; then
+  echo "No build.gradle.kts found"
+  exit 1
+fi
+
+remote=$GRADLE_REMOTE_SSH
+if [ -z "$remote" ]; then
+  echo "$GRADLE_REMOTE_SSH is not set, please set it to a valid SSH login on your build server"
+  exit 1
+fi
+
+remote_dir=$GRADLE_REMOTE_DIR
+if [ -z "$remote_dir" ]; then
+  echo "$GRADLE_REMOTE_DIR is not set, please set it to the directory where the project should be copied on the build server"
+  exit 1
+fi
+
+remote_project_dir="$GRADLE_REMOTE_DIR/$(basename "$(dirname "$PWD")")_$(basename "$PWD")"
+
+set -e
+
+# Copy the project to the remote
+rsync -irci --exclude build --exclude .gradle --include .gradle/gradle.properties --delete . "$remote:$remote_project_dir"
```

Comment:
I think we should go ahead and exclude `.signing` to avoid accidental footguns. We can't protect all cases, but this one would be pretty important.

You could also consider using `git ls-files` with options to include only tracked files, the developer would have to git add the files if they want to transfer them.

Reply:
Adding `.signing` makes total sense. I updated the script to include an array for exclusions and inclusions.

`git ls-files` would make sense based on the use case for the script. At the moment I use it to offload builds to another machine under my control. So risk is minimal, but when this script is used to offload builds to a remoter server it needs further mitigations.